### PR TITLE
CRIU invokes j9sysinfo_get_env() to avoid checkpoint failure

### PR DESCRIPTION
**Pull request by JasonFengJ9**
CRIU invokes `j9sysinfo_get_env()` to avoid checkpoint failure

When `SSSD` is enabled, `j9sysinfo_get_username`/`getpwuid` can cause checkpoint failure, these methods will be skipped if a checkpoint can be taken, `j9sysinfo_get_env` is used instead;
Add `sysinfoGetUserNameHelper` to wrap `j9sysinfo_get_username()` and `j9sysinfo_get_env()` calls;
Add `J9PORT_CTLDATA_INSTANTON_FLAGS` to set `CRIU` status enabled & final restore via` j9port_control` accordingly.

FYI @hangshao0 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: runtime/port/sysvipc/j9shmem.c [hunk 1]
```diff
@@ -1290,29 +1290,30 @@ j9shmem_getDir(struct J9PortLibrary* portLibrary, const char* ctrlDirName, uint3
...
+						homeDir = pwent->pw_dir;
 					} else {
-						rc = J9PORT_ERROR_SHMEM_GET_DIR_FAILED_TO_GET_HOME;
-						Trc_PRT_j9shmem_getDir_tryHomeDirFailed_getpwuidFailed();
+						rc = J9PORT_ERROR_SHMEM_GET_DIR_HOME_BUF_OVERFLOW;
+						Trc_PRT_j9shmem_getDir_tryHomeDirFailed_pw_dirDirTooLong(dirLen, minBufLen - baseDirLen);
 					}
+				} else {
+					rc = J9PORT_ERROR_SHMEM_GET_DIR_FAILED_TO_GET_HOME;
+					Trc_PRT_j9shmem_getDir_tryHomeDirFailed_getpwuidFailed();
```
Comment (hangshao0) [inline runtime/port/sysvipc/j9shmem.c:1316]: This trace point is `TraceException`, should it be recoredd as an exception when `NULL == portLibrary->finalRestore` when `getpwuid()` is not called ? And the trace message "getpwuid() returns NULL" is not accurate anymore in this case.

File: runtime/shared_common/shrinit.cpp [hunk 2]
```diff
@@ -2624,6 +2625,44 @@ performSharedClassesCommandLineAction(J9JavaVM* vm, J9SharedClassConfig* sharedC
...
+	if (!vmFuncs->isCheckpointAllowed(vmFuncs->currentVMThread(vm)))
+#endif /* defined(J9VM_OPT_CRIU_SUPPORT) */
+	{
+		rc = j9sysinfo_get_username(buffer, length);
+		if (rc > 0) {
+			SHRINIT_ERR_TRACE2(verboseFlags, J9NLS_SHRC_SHRINIT_FAILURE_COPYING_USERNAME_TOOLONG, length, rc);
+		}
+	}
+	if (rc < 0) {
+		rc = j9sysinfo_get_env("USER", buffer, length);
```
Comment (hangshao0) [inline runtime/shared_common/shrinit.cpp:2655]: For the user name, we are trying `getpwuid()` first, then the environment variable, which is inconsistent with `j9shmem_getDir()` where the environment variable is queried first.

File: runtime/port/sysvipc/j9shmem.c (hunk 1 above)
Comment (JasonFengJ9) [reply runtime/port/sysvipc/j9shmem.c]: Added `Trc_PRT_j9shmem_getDir_tryHomeDirFailed_notFinalRestore()` for the case `finalRestore` is `FALSE`.

File: runtime/shared_common/shrinit.cpp (hunk 2 above)
Comment (JasonFengJ9) [reply runtime/shared_common/shrinit.cpp]: Switched the order within `sysinfoGetUserNameHelper()`, attempt `j9sysinfo_get_env()` first.

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
jenkins test sanity win jdk8

**Comment by tajila** [comment]
Jenkins test sanity xlinuxcriu jdk17

**Pull request merged**